pyproj/utils.py

```diff
@@ -30,7 +30,14 @@ def _copytobuffer(xx: Any) -> Tuple[Any, bool, bool, bool]:
     isfloat = False
     islist = False
     istuple = False
-    # first, if it's a numpy array scalar convert to float
+    # check for pandas.Series, xarray.DataArray or dask.array.Array
+    if hasattr(xx, "__array__") and callable(xx.__array__):
+        try:
+            xx = xx.__array__(dtype="d")
```

Comment:
What does `dtype="d"` do here? Give you a byte array?

Reply:
It converts to a double-like:
```python
>>> import xarray
>>> da = xarray.DataArray([1])
>>> da.dtype
dtype('int64')
>>> da.__array__(dtype="d")
array([1.])
>>> da.__array__(dtype="d").dtype
dtype('float64')
```

Reply:
I just removed it as the `dtype` kwarg isn't always there for all methods. Actually, I guess it is just an arg for numpy:
```python
>>> na = numpy.array([1])
>>> na.dtype
dtype('int64')
>>> na.__array__("d")
array([1.])
>>> na.__array__("d").dtype
dtype('float64')
>>> na.__array__(dtype="d").dtype
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __array__() takes no keyword arguments
```

```
__array__(...) method of numpy.ndarray instance
    a.__array__(|dtype) -> reference if type unchanged, copy otherwise.
```

Reply:
Either way, it does not need it as `astype` is called later.